src/libraries/System.Private.Xml/tests/XmlReader/Tests/ReadCharsTests.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Text;
+using Xunit;
+
+namespace System.Xml.XmlReaderTests
+{
+    public class ReadCharsTests
+    {
+        [Fact]
+        public static void ReadCharsInLoop()
+        {
+            string xml = GenerateTestXml(out string expected);
+
+            // Load the reader with the xml.  Ignore white space.
+            using (XmlTextReader reader = new XmlTextReader(new StringReader(xml)))
+            {
+                reader.WhitespaceHandling = WhitespaceHandling.None;
+                reader.DtdProcessing = DtdProcessing.Ignore;
+
+                // Set variables used by ReadChars.
+                int charbuffersize = 10;
+                char[] buffer = new char[charbuffersize];
+
+                // Parse the xml.  Read the element content
+                // using the ReadChars method.
+                reader.MoveToContent();
+                StringBuilder sb = new StringBuilder();
+                int iCnt;
+                while ((iCnt = reader.ReadChars(buffer, 0, charbuffersize)) > 0)
+                {
+                    // Append the buffer contents.
+                    sb.Append(buffer, 0, iCnt);
+                    // Clear the buffer.
+                    Array.Clear(buffer, 0, charbuffersize);
+                }
+
+                Assert.Equal(expected, sb.ToString());
+            }
+        }
+
+        private static string GenerateTestXml(out string expected)
+        {
```

Comment:
consider just using const and and possibly string.Replace for line ending

Reply:
I've found this way of generating test xmls in here: https://github.com/dotnet/runtime/blob/8de1809155ea6c32494d5c5816455c7f332ea147/src/libraries/System.Private.Xml/tests/Readers/NameTable/TestFiles.cs#L80, and decided that this is the preferred way in this repo. 
If you like - I can change it.

Reply:
this code base is old and ugly, let's at least make new code prettier since we don't have time on improving the entire code base